core/src/main/java/com/linecorp/armeria/common/AbstractRequestContext.java

```diff
@@ -180,4 +208,10 @@ public final int hashCode() {
     public final boolean equals(Object obj) {
         return super.equals(obj);
     }
+
+    private void cancelIfTimedOut() {
+        if (timedOut) {
+            throw new CancellationException("Context timed out, skipping callback invocation.");
```

Comment:
Could you pre-instantiate the exception with empty stack trace when `com.linecorp.armeria.common.util.Exceptions.isVerbose()` is `false`?